### PR TITLE
Edit Dossier comment/note implementation in an overlay (link in byline)

### PR DESCRIPTION
**Pull request by maethu**
We need to discuss the following:
- [x] ~Where should the button be? (IMHO the byline is perfect)~ 👍 
- [x] ~Is a info message needed (after saving or abording)?~: yes please.
- [x] ~Does the note overlay ned a title?~: {Dossier-Title} Kommentar:
- [x] Can I use the plone.restapi?: No.
- [x] I checked the Controller.js of @bierik  but it seems not to suit my needs, I'll have look with him.
- [x] ~Tests are failing, but IMHO not my fault 😏 :~ only one test fails, and it is your fault 😝 
- [x] ~Trix (Ja/Nein)~?: Not yet.
- [x] ~Moves styles / ressources~
- [x] How to indicate, that there is no comment? How to indicate, that there is a comment?
- [x] Based on https://github.com/4teamwork/plonetheme.teamraum/pull/500 (pls merge first)
**Testing:**
-  [x] Are going to be implemented after all decision have been made. Currently the code ist only a poc. 

gever_note

### CODE REVIEW

**Comment by maethu** [comment]
@deiferni 
I fixed some technical points. 

gever_note

**Comment by bierik** [comment]
Why do we not use the approach from the meeting here https://github.com/4teamwork/opengever.core/pull/2526/commits/1a140a1d7d760b0ce935e44e68fb5839520fd1e7. What happens when the internet connection accidentally cuts off.

**Comment by maethu** [comment]
@bierik We need to check the integration later today. But you were not here yesterday 😏

**Comment by maethu** [comment]
@deiferni This one is now ready, only the  4teamwork/plonetheme.teamraum#500 is left!

**Comment by deiferni** [comment]
@maethu could you squash the commits before we review again? Currently there is a lot of noise, and stuff being added and removed again, this makes it hard do review commits separately.

**Comment by maethu** [comment]
@deiferni updated.

**Comment by maethu** [comment]
@deiferni I removed the commit, which checked out the theme branch.

File: opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po [hunk 1]
```diff
@@ -400,6 +405,16 @@ msgstr "Schlagwörter zur Umschreibung eines Dossiers. Nicht zu verwechseln mit
 msgid "help_number_of_containers"
 msgstr "Anzahl Behälter, die ein (grosses) Dossier in Papierform enthalten"
 
+#. Default: "Create a new note"
+#: ./opengever/dossier/viewlets/templates/note.pt:18
+msgid "help_text_add_note"
+msgstr "Neuer Kommentar erfassen."
```
Comment (deiferni) [inline opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po]: Neuen...

File: opengever/dossier/tests/test_dossier_byline.py [hunk 2]
```diff
@@ -94,6 +96,81 @@ def test_dossier_byline_hide_filing_number(self, browser):
...
+        self.assertEquals(['Edit Note', 'Add Note'],
+                          comment.css('#editNoteLink .linkLabel').text)
+
+    @browsing
+    def test_dossier_byline_show_comments_editlink_on_maindossier_only(
+            self, browser):
+        browser.login().visit(self.dossier)
+        comment = self.get_byline_value_by_label('Dossiernote:')
+        self.assertTrue(comment,
+                        'Exepct the edit comment link in dossier byline')
```
Comment (deiferni) [inline opengever/dossier/tests/test_dossier_byline.py]: typo Exepct

File: opengever/dossier/tests/test_dossier_byline.py [hunk 3]
```diff
@@ -94,6 +96,81 @@ def test_dossier_byline_hide_filing_number(self, browser):
...
+
+        subdossier = create(Builder('dossier').within(self.dossier))
+        browser.visit(subdossier)
+        comment = self.get_byline_value_by_label('Dossiernote:')
+        self.assertFalse(comment,
+                         'Exepct NO edit comment link on subdossier')
```
Comment (deiferni) [inline opengever/dossier/tests/test_dossier_byline.py]: typo Exepct

File: opengever/dossier/tests/test_dossier_byline.py [hunk 4]
```diff
@@ -94,6 +96,81 @@ def test_dossier_byline_hide_filing_number(self, browser):
...
+    def test_dossier_byline_hide_comments_editlink_without_modify_rights(
+            self, browser):
+        self.dossier.manage_permission('Modify portal content',
+                                       roles=[],
+                                       acquire=False)
+        transaction.commit()
+        browser.login().visit(self.dossier)
+        comment = self.get_byline_value_by_label('Dossiernote:')
+        self.assertFalse(comment,
+                         'Exepct NO edit comment link on dossier')
```
Comment (deiferni) [inline opengever/dossier/tests/test_dossier_byline.py]: Exepct

File: opengever/dossier/profiles/default/jsregistry.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<object name="portal_javascripts" meta_type="JavaScripts Registry" autogroup="False">
+
+  <javascript cacheable="True" compression="safe" cookable="True" enabled="on"
```
Comment (deiferni) [inline opengever/dossier/profiles/default/jsregistry.xml]: When i install a new plone-site i get js-errors, could you double-check js-registry insertion order?

screen shot 2017-02-09 at 14 40 57

File: opengever/dossier/viewlets/byline.py [hunk 6]
```diff
@@ -21,17 +27,38 @@ def end(self):
         dossier = IDossier(self.context)
         return self.to_localized_time(dossier.end)
 
+    note_template = ViewPageTemplateFile('templates/note.pt')
+
+    def note(self):
+        dossier = IDossier(self.context)
+        msg_confirm = _(u'note_text_confirm_abbord', default=u'Confirm abbord')
```
Comment (deiferni) [inline opengever/dossier/viewlets/byline.py]: abort

File: opengever/dossier/viewlets/byline.py [hunk 7]
```diff
@@ -21,17 +27,38 @@ def end(self):
...
+        msg_title_info = ogbmf(u'message_title_info', default=u'Information')
+        msg_title_error = ogbmf(u'message_title_error', default=u'Error')
+        msg_body_info = _(u'message_body_info', default=u'Changes saved')
+        msg_body_error = _(u'message_body_error', default=u'Changes not saved')
+        create_translations = lambda msg: translate(msg, context=self.request)
+
+        translations = json.dumps(
+            {'note_text_confirm_abbord': create_translations(msg_confirm),
```
Comment (deiferni) [inline opengever/dossier/viewlets/byline.py]: abort

File: opengever/dossier/viewlets/templates/note.pt [hunk 8]
```diff
@@ -0,0 +1,47 @@
+<span class="editNoteWrapper"
+      i18n:domain="opengever.dossier"
+      tal:define="has_notes options/note">
+  <a href="#" id='editNoteLink'
+     tal:attributes="data-notecache string:${options/note};
+                     data-i18n string:${options/translations}">
+    <span tal:define="hide_class python: not has_notes and 'hide' or ''"
```
Comment (deiferni) [inline opengever/dossier/viewlets/templates/note.pt]: python has a proper ternary `"" if has_notes else "hide"` please use that one

File: opengever/dossier/viewlets/templates/note.pt [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+          tal:attributes="class string:edit ${hide_class}">
+      <span class="linkLabel" i18n:translate="label_edit_note">Edit Note</span>
+      <span class="visualHint"
+            title="There is a note"
+            i18n:attributes="title help_text_edit_note">!</span>
+    </span>
+
+    <span tal:define="hide_class python: has_notes and 'hide' or ''"
```
Comment (deiferni) [inline opengever/dossier/viewlets/templates/note.pt]: ternary

File: opengever/dossier/profiles/default/jsregistry.xml (hunk 5 above)
Comment (maethu) [inline opengever/dossier/profiles/default/jsregistry.xml]: I also had a issue for newly created sites... I've fixed that. 

But the second failure is not mine 😼 
screen shot 2017-02-09 at 15 51 59

Comment (maethu) [inline opengever/dossier/profiles/default/jsregistry.xml]: Open new issue: https://github.com/4teamwork/opengever.core/issues/2545

**Pull request merged**